### ISSUE TITLE
Test executed by workers in multiple browsers unexpectedly fails: -1 failed

### ISSUE THREAD

**Issue by mirao** [codeceptjs/CodeceptJS#2790]
`$ n run 12 bin/parallel.js` (NodeJS is started using https://www.npmjs.com/package/n)
#### What are you trying to achieve?
```bash
Passed :  Check Adminer
Passed :  Check Adminer
[2] OK  | 2 passed  // 3888s
```
#### What do you get instead?

> Provide console output if related. Use `--verbose` mode for more details.

```bash
Passed :  Check Adminer
Passed :  Check Adminer
[3]
[3]   FAIL  | 2 passed, -1 failed   // 3888s
[3]
[3] -- FAILURES:
```

> Provide test source code if related

tests/my/My_test.js:
```js
Feature("My");

Scenario("Check Adminer", ({ I }) => {
    I.amOnPage("/");
    I.see("Adminer");
});
```
bin/parallel.js:
```js
const { Workers, event } = require("codeceptjs");
// here will go magic

const workerConfig = {
    testConfig: "codecept.conf.js",
};

// don't initialize workers in constructor
const workers = new Workers(null, workerConfig);
// split tests by suites in 2 groups
const testGroups = workers.createGroupsOfSuites(2);

/** Browser configuration for run-workers mode */
const browsers = [
    {
        port: 4444,
        browser: "chrome",
    },
    {
        port: 4446,
        browser: "firefox",
    }
];

const configs = browsers.map(browser => {
    return {
        helpers: {
            WebDriver: browser
        }
    };
});

for (const config of configs) {
    for (const group of testGroups) {
        const worker = workers.spawn();
        worker.addTests(group);
        worker.addConfig(config);
    }
}

// Listen events for failed test
workers.on(event.test.failed, (failedTest) => {
    console.log("Failed : ", failedTest.title);
});

// Listen events for passed test
workers.on(event.test.passed, (successTest) => {
    console.log("Passed : ", successTest.title);
});

// test run status will also be available in event
workers.on(event.all.result, () => {
    // Use printResults() to display result with standard style
    workers.printResults();
});

async function runWorkers() {
    try {
        // run bootstrapAll
        await workers.bootstrapAll();
        // run tests
        await workers.run();
    } finally {
        // run teardown All
        await workers.teardownAll();
    }
}

// run workers as async function
runWorkers();

```
### Details

* CodeceptJS version: 3.0.4
* NodeJS Version: v12.20.1 (using https://www.npmjs.com/package/n)
* Operating System: Ubuntu 20.10
* webdriverio 6.6.2
* Chrome/Firefox running in own docker container with Selenium
* Configuration file:

```js
exports.config = {
    tests: "./*_test.js",
    output: "./output",
    helpers: {
        WebDriver: {
            url: "http://172.50.0.1:8080/adminer",
            browser: "chrome"
        }
    },
    include: {
        I: "./steps_file.js"
    },
    bootstrap: null,
    mocha: {},
    name: "my",
    plugins: {
        pauseOnFail: {},
        retryFailedStep: {
            enabled: true
        },
        tryTo: {
            enabled: true
        },
        screenshotOnFail: {
            enabled: true
        }
    }
};

```

**Comment by mirao** [comment]
I checked that if I run "parallel" script with configuration for one browser only, e.g.:
```js
const browsers = [
    // {
    //     port: 4444,
    //     browser: "chrome",
    // },
    {
        port: 4446,
        browser: "firefox",
    }
];
```
Then test passes:
``` bash
$ n run 12 bin/parallel.js 
Passed :  Check Adminer
[1]
[1]   OK  | 1 passed   // 3984s
```

**Comment by mirao** [comment]
When the test contains two scenarios instead of one, I get "-2 failed" (for configuration with two browsers):
```
$ n run 12 bin/parallel.js 
Passed :  Check Adminer
Passed :  Check Adminer
Passed :  Check Another
Passed :  Check Another
[3]
[3]   FAIL  | 4 passed, -2 failed   // 257423s
[3]
[3] -- FAILURES:
```

**Comment by PeterNgTr** [comment]
I guess this was an issue that due to duplicated test scenarios. May you try running tests again without any duplications? https://github.com/codeceptjs/CodeceptJS/pull/2656

**Comment by mirao** [comment]
@PeterNgTr No, the original "tests/my/My_test.js" is the only suite in the project with only scenario. So there is no duplicate.

**Comment by PeterNgTr** [comment]
@mirao even thou you have one test but you run in parallel in two browsers -> 2 tests and they are the same name, I believe. Trying to pass the browser to test name :)

**Comment by mirao** [comment]
It seems to be a duplicate of #2158.

**Comment by kobenguyent** [comment]
Closed for now! Feel free to reopen if you still encounter the issue with latest version and would be nice to provide a sample code to reproduce the issue.
